### PR TITLE
 Added Cypress test for validating MM-18043 Guest User Experience: Verify Guest User Restrictions

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
Validating the Restrictions when a Guest User logs in. Please check the ticket for more details.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-18043

### CODE REVIEW

File: e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        cy.get('#channelHeaderDescription').within(($el) => {
+            cy.wrap($el).find('.has-guest-header').should('be.visible').and('have.text', 'This channel has guests');
+        });
+
+        // * Verify list of Users and Guest Badge in Channel Members List
+        cy.get('#member_popover').click();
+        cy.get('#member-list-popover').should('be.visible').within(($el) => {
+            cy.wrap($el).find('.more-modal__body > div').
+                children().should('have.length', 2);
+            cy.wrap($el).find('.Badge').should('be.visible').and('have.text', 'GUEST');
```
Comment (saturninoabril) [inline e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js]: There are only 2 users. Can we verify which user has a badge or not?

File: e2e/cypress/support/api_commands.js [hunk 2]
```diff
@@ -666,3 +666,41 @@ Cypress.Commands.add('apiGetTeam', (teamId) => {
...
+                url: '/api/v4/users/login',
+                method: 'POST',
+                body: {login_id: newUser.username, password: newUser.password},
+            }).then(() => {
+                cy.visit('/');
+                return cy.wrap(newUser);
+            });
+        });
+    });
+});
```
Comment (saturninoabril) [inline e2e/cypress/support/api_commands.js]: 💯

File: e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js (hunk 1 above)
Comment (srkgupta) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js]: Ok sure. Let me check and do the needful changes

Comment (srkgupta) [reply e2e/cypress/integration/enterprise/guest_accounts/guest_experience_ui_spec.js]: Hi @saturninoabril 
Made the necessary changes

**Comment by prapti** [comment]
Tests for guest user are failing. @srkgupta is looking into it. It will be tested again once resolved.

**Comment by srkgupta** [comment]
Deleting this PR and creating a new PR with fresh commits.

**Pull request closed without merging**